src/popup/pages/TransactionDetails.vue

```diff
@@ -313,7 +313,7 @@ export default defineComponent({
       outerTxType,
     } = useTransactionTx({
       store: root.$store,
-      externalAddress: props.transactionOwner,
+      externalAddress: props.transactionOwner || root.$route?.query?.owner as string,
```

Comment:
I think instead of passing this alternative value by query the logic that was set in the `TransactionItem` file can be moved before the `useTransactionTx` in this file.

Reply:
Without query on refresh we wont be able to know which we have loaded.

Reply:
Yeah, I get this. But doesn't that create a vulnerability? I can send you a link to my transaction where you will be shown as the owner. So I still think the logic for determining who is the owner should be encapsulated within this page.
Or I'm missing something?